packages/sdk/src/types/attributes/IQLQuery.ts

```diff
@@ -1,5 +1,5 @@
 export interface IQLQuery {
-    "@type"?: "IQLQuery";
+    "@type": "IQLQuery";
```

Comment:
As Julian asked in the tests, why no longer optional?

Reply:
Hi this is a little bit irritating for me.

https://github.com/nmshd/runtime/blob/39b5138435a9b7e137a36d21278c76fa81e0fe33/packages/content/src/attributes/IQLQuery.ts#L33

Here it also is not optional anymore. But in the `from` method the property is omitted

https://github.com/nmshd/runtime/blob/39b5138435a9b7e137a36d21278c76fa81e0fe33/packages/content/src/attributes/IQLQuery.ts#L59

Reply:
When I try to call the runtime use case and omit the `"@type": "IQLQuery"` i will get an error that it could not be parsed

Reply:
> When I try to call the runtime use case and omit the "@type": "IQLQuery" i will get an error that it could not be parsed
Ok, that's strange ...

- As we directly deserialize via IQLQuery.from with the Runtime's Usecase (see https://github.com/nmshd/runtime/blob/main/packages/runtime/src/useCases/consumption/attributes/ExecuteIQLQuery.ts#L28 ) we do not need @type: "IQLQuery" within the JSON for any payload to POST .../ExecuteIQLQuery (as ts-serval is automatically using the default type then)
- I guess this is caused by a TypeScript type error b/c the interface of ExecuteIQLQueryRequest seems to be wrong, as it should be possible to omit the @ type definition in the request (because of the first bullet point): https://github.com/nmshd/runtime/blob/main/packages/runtime/src/useCases/consumption/attributes/ExecuteIQLQuery.ts#L10
- I can remember that @slavistan had some problems when building the schemas within the Runtime because of some ugly bugs which might be a reason/cause of not omitting the @ type within the ExecuteIQLQueryRequest interface in the Runtime.
- BUT: I can imagine that using the Connector however all works out good, as all the requests coming in are any's and thus the TypeScript error is ommitted.
- Thus: I can imagine that using the SDK without the @ type: "IQLQuery" definition is working great, and we could remove this change. But we still have an inconsistency within the Runtime's request types of the Usecases.


Furthermore:
- Compared to RelationshipAttributeQuery, we're doing exact the same thing within the Runtime (the @ type is defined within the RelationshipAttributeQuery): https://github.com/nmshd/runtime/blob/39b5138435a9b7e137a36d21278c76fa81e0fe33/packages/content/src/attributes/RelationshipAttributeQuery.ts#L69
- BUT within the type definition of RelationshipAttributeQueryJSON in the Connector SDK, the @ type is optional: https://github.com/nmshd/connector/blob/main/packages/sdk/src/types/attributes/RelationshipAttributeQuery.ts#L2
- So as we're doing the same thing, using the Runtime Usecase ExecuteRelationshipAttributeQuery without an @ type:"RelationshipAttributeQuery" is also doing fine?
- And we might have the same Runtime request type issues for Execute... (or even more?)

Reply:
I believe the error occurs during validation of the UseCase: https://github.com/nmshd/runtime/blob/main/packages/runtime/src/useCases/consumption/attributes/ExecuteIQLQuery.ts#L15, as the schema of "ExecuteIQLQueryRequest" requires @type to be defined:
https://github.com/nmshd/runtime/blob/main/packages/runtime/src/useCases/common/Schemas.ts#L15948
This validation seems to be missing in the IdentityAttributeQuery: https://github.com/nmshd/runtime/blob/main/packages/runtime/src/useCases/consumption/attributes/ExecuteIdentityAttributeQuery.ts#L15

Reply:
I'll wait for this to be merged and will update the code

Reply:
is optional again

Reply:
I've gone a quite different route - recreating what we have in the actual types. Now we can omit it when KNOWING what kind of query we have.